### PR TITLE
docs: update grid cell focus example to use global styles

### PR DESCRIPTION
**Pull request by sissbruecker**
Updates the grid cell focus example to use the new theming mechanism.

### CODE REVIEW

File: frontend/themes/docs/grid-cell-focus.css [hunk 1]
```diff
@@ -1,6 +1,7 @@
-/* frontend/theme/[my-theme]/components/vaadin-grid.css */
+/* Add this to your global CSS, for example in: */
+/* frontend/theme/[my-theme]/styles.css */
 
-:host([theme~='force-focus-outline']) [part~='cell']:focus::before {
+vaadin-grid.force-focus-outline::part(cell):focus::before {
```
Comment (web-padawan) [inline frontend/themes/docs/grid-cell-focus.css:4]: Let's use `::part(focused-cell):focus` instead as we now have this "cell button" mode for VoiceOver on Mac OS.

**Review by web-padawan — CHANGES_REQUESTED** [review]

File: frontend/themes/docs/grid-cell-focus.css (hunk 1 above)
Comment (sissbruecker) [reply frontend/themes/docs/grid-cell-focus.css:4]: Thanks, though it should be `::part(focused-cell)::before`, correct?

Comment (web-padawan) [reply frontend/themes/docs/grid-cell-focus.css:4]: Yes, indeed 👍

Comment (sissbruecker) [reply frontend/themes/docs/grid-cell-focus.css:4]: Done

**Review by web-padawan — APPROVED** [review]

File: frontend/themes/docs/grid-cell-focus.css [hunk 2]
```diff
@@ -1,6 +1,7 @@
-/* frontend/theme/[my-theme]/components/vaadin-grid.css */
+/* Add this to your global CSS, for example in: */
+/* frontend/theme/[my-theme]/styles.css */
 
-:host([theme~='force-focus-outline']) [part~='cell']:focus::before {
+vaadin-grid.force-focus-outline::part(focused-cell)::before {
```
Comment (vursen) [inline frontend/themes/docs/grid-cell-focus.css:4]: Why do we prefer classes in this case? To me, it feels like a theme.

Comment (vursen) [reply frontend/themes/docs/grid-cell-focus.css:4]: Well, perhaps we can use classes if this customization only applies to this particular example.

Comment (web-padawan) [reply frontend/themes/docs/grid-cell-focus.css:4]: Yes, from now on we should use CSS classes for any individual customizations like this one.

**Review by vursen — APPROVED** [review]

**Pull request merged**